src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -353,6 +412,13 @@ public ISymbolNode ExternSymbol(string name)
             return _externSymbols.GetOrAdd(name);
         }
 
+        private NodeCache<MethodDesc, ExternMethodSymbolNode> _externMethodSymbols;
+
+        public ISymbolNode ExternSymbol(MethodDesc method)
+        {
+            return _externMethodSymbols.GetOrAdd(method);
```

Comment:
I don't think we should have this flavor of `ExternSymbol` as a public method. Can you inline this into the place where it's used?